### ISSUE TITLE
[Training] Potential shape bug: `del labels_input.type.tensor_type.shape.dim[-1]` assumes channel-last format in loss.py

### ISSUE THREAD

**Issue by StucchiD** [microsoft/onnxruntime#25792]
### Describe the issue

I am trying to create the ONNX evaluation and training graphs starting from a pytorch model. The issue is with the labels shape inferred automatically by the CrossEntropyLoss build method.

The change on [this line](https://github.com/microsoft/onnxruntime/commit/6236707c640b84857f9e64c1c43fb9f304ca7749#diff-551788247d076b48e3f3e25b5b347fb7c05a04955c013a3da1eb3bce8f76d5e8R99) assumes that data format is channel-last. This should not work, for example, in a per-pixel loss where the last activation map is (batch_size, out_channels, width, height), where the height dimension whould be deleted.

Is this correct? Since this is my first opened issue, I do not know how should I act in this case.

### To reproduce

Pytorch model:
```python
class SimpleConvModel(nn.Module):
    def __init__(self):
        super(SimpleConvModel, self).__init__()
        self.conv = nn.Conv2d(
            in_channels=3, out_channels=5,
            kernel_size=3, stride=1)

    def forward(self, x):
        x = self.conv(x)
        return x
```

ONNX Model Training block used to create the backward graph from the model
```python
class ModelTrainingBlock(onnxblock.TrainingBlock):
    def __init__(self):
        super(ModelTrainingBlock, self).__init__()
        self.loss = onnxblock.loss.CrossEntropyLoss()

    def build(self, output_name):
        return self.loss(output_name), output_name
```

I use this block to create the eval and training graphs, as follows:
```python
with onnxblock.base(onnx_model):
    _ = training_block(*[output.name for output in onnx_model.graph.output])
    training_model, eval_model = training_block.to_model_proto()
```

The labels shape automatically inferred is wrong, as it removes the last dimension.

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

onnxruntime==1.18.1, onnxruntime-trainining==1.19.2

### PyTorch Version

2.7.1

### Execution Provider

Default CPU
